liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -299,6 +286,34 @@ public void run() {
       }
    }
 
+    /***
+     * Initializes a thread as a shutdown-hook to run <i>bin/server stop</i> shell
+     * script command upon JVM shut-down.
+     *
+     * Utilizes the initial command list for startup by just replacing start-up command (run)
+     * with shut-down (stop).
+     *
+     * @param cmd
+     * @return an initialized unstarted thread to be registered as a shutdown hook.
+     */
+    private Thread getShutDownThread(final List<String> cmd) {
+       Collections.replaceAll(cmd, "run", "stop");
```

Comment:
This seems a bit hacky to me. I think it's preferable to refactor the command definition that is passed to the `ProcessBuilder` (lines 202 to 211) and pass in a flag that indicates whether this is a `run` or `stop` command.

Reply:
Please review 280abcd. Tried to refactor some stuff, in order to make the code look less "hacky" whilst avoiding over-engineering a seemingly simple branching.

Do let me know if you have any other inputs =)

Reply:
Looks good, thanks!